src/component/Chart.jsx

```diff
@@ -0,0 +1,317 @@
+                        scaleKey,
+                        scaleTypeKey,
+                        columns: childrenGroups
+                            .get('canvas')
+                            .map(ii => List([].concat(ii[dimensionKey]).concat(props[dimensionKey])))
+                            .flatten(1)
+                            .toSet()
+                            .filter(ii => ii)
+                            .toArray()
+                    };
+                })
+        };
+
+        this.getChildProps = this.getChildProps.bind(this);
+        this.getCanvasSize = this.getCanvasSize.bind(this);
+        this.getAxisSize = this.getAxisSize.bind(this);
+    }
+    getCanvasSize(): Object {
+        const {width, height, padding = []} = this.props;
+        // $FlowBug: flow cant handle default assignment on array destructuring
+        const [top = 0, right = 0, bottom = 0, left = 0] = padding;
+
+        return {
+            width: Math.max(width - left - right, 0), // clamp negatives
+            height: Math.max(height - top - bottom, 0), // clamp negatives
+            outerWidth: width,
+            outerHeight: height,
+            top,
+            right,
+            bottom,
+            left
+        };
+    }
+    getAxisSize(axisType: string): Object {
+        const {top, right, bottom, left, width, height} = this.getCanvasSize();
+        switch(axisType) {
+            case 'top':
+                return {
+                    width,
+                    height: top
+                };
+
+            case 'right':
+                return {
+                    width: right,
+                    height
+                };
+
+            case 'bottom':
+                return {
+                    width,
+                    height: bottom
+                };
+
+            case 'left':
+                return {
+                    width: left,
+                    height
+                };
+
+            default:
+                return {};
+
+        }
+    }
+    getDefaultScale(dimension: Object): Function {
+        const {dimensionName, scaleTypeKey, dimensionKey, columns} = dimension;
+
+        switch(dimensionName) {
+            case 'x':
+            case 'y':
+                return (pp: Object): Function => {
+                    // choose the max value of range based on x/width y/height
+                    const bound = (dimensionName === 'x') ? pp.width : pp.height;
+                    // Make the current dimension always a list
+                    const currentDimension = List().concat(pp[dimensionKey]);
+
+                    if(pp[dimensionKey] === undefined) {
+                        throw new Error(`a ${pp.chartType} child did not choose a ${dimensionKey} and the chart has not provided it.`);
+                    }
+
+
+                    // create a set of booleans to check if a group is mixing dimension types
+                    function isContinuous(list: List): Set {
```

Comment:
Looks good, wouldn't mind moving some of this logic into ChartData so the usage here is cleaner, and give it memoization etc. Something like `chartData.doesMyColumnListContainMixedContinuityApricots()`.